specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,422 @@
+  /// Did not perform Save As because the end user cancelled or the 
+  /// CoreWebView2SaveAsRequestedEventArgs.Cancel property was set to TRUE.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_CANCELLED,
+  /// The RequestSaveAs method call completed successfully and the download has started.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_COMPLETED,
+} COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS;
+
+[uuid(15e1c6a3-c72a-4df3-91d7-d097fbec3bfd), object, pointer_default(unique)]
+interface ICoreWebView2_20 : IUnknown {
+  /// Programmatically trigger a save as action for the current top-level document. 
+  /// The `SaveAsRequested` event will be raised.
+  ///
+  /// Opens a system modal dialog by default. If it was already opened, this method 
+  /// would not open another one. If the `SuppressDefaultDialog` is TRUE, won't open 
+  /// the system dialog.
+  ///
+  /// The method can return a detailed info to indicate the call's result. 
+  /// Please see COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS
+  ///
+  /// \snippet ScenarioSaveAs.cpp ProgrammaticSaveAs
+  HRESULT RequestSaveAs([in] ICoreWebView2RequestSaveAsCompletedHandler* handler);
+
+  /// Add an event handler for the `SaveAsRequested` event. This event is raised
+  /// when save as is triggered, programmatically or manually.
+  ///
+  /// \snippet ScenarioSaveAs.cpp AddEventHandler
+  HRESULT add_SaveAsRequested(
+   [in] ICoreWebView2SaveAsRequestedEventHandler* eventHanlder,
+   [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_SaveAsRequested`.
+  HRESULT remove_SaveAsRequested(
+   [in] EventRegistrationToken token);
+}
+
+/// The event handler for the `SaveAsRequested` event.
+[uuid(55b86cd2-adfd-47f1-9cef-cdfb8c414ed3), object, pointer_default(unique)]
+interface ICoreWebView2SaveAsRequestedEventHandler : IUnknown {
+  HRESULT Invoke(
+   [in] ICoreWebView2* sender,
+   [in] ICoreWebView2SaveAsRequestedEventArgs* args);
+}
+
+/// The event args for `SaveAsRequested` event
+[uuid(80101027-b8c3-49a1-a052-9ea4bd63ba47), object, pointer_default(unique)]
+interface ICoreWebView2SaveAsRequestedEventArgs : IUnknown {
+  /// Get the Mime type of content to be saved
+  [propget] HRESULT ContentMimeType([out, retval] LPWSTR* value);
+
+  /// You can set this to TRUE to cancel the Save As. Then the download won't start. 
+  /// A programmatic call will return COREWEBVIEW2_SAVE_AS_CANCELLED as well. 
+  ///
+  /// The default value is FALSE.
+  ///
+  /// Set the `Cancel` for save as
+  [propput] HRESULT Cancel ([in] BOOL value);
+
+  /// Get the `Cancel` for save as
+  [propget] HRESULT Cancel ([out, retval] BOOL* value);
+
+  /// Indicates if the system default dialog will be supressed, FALSE means
+  /// save as default dialog will show; TRUE means a silent save as, will
+  /// skip the system dialog. 
+  ///
+  /// The default value is FALSE.
+  ///
+  /// Set the `SupressDefaultDialog`
+  [propput] HRESULT SupressDefaultDialog([in] BOOL value);
```

Comment:
If I set both SuppressDefaultDialog=true and Cancel=true, I effectively disable Save As.  
If I set Cancel=true but don't set SuppressDefaultDialog, will the default dialog show up to try its own operation or will the Cancel win?  
If I "handle" the SaveAs request and populate args Kind and ResultFilePath, but don't set SuppressDefaultDialog=true, does the system ignore my other args values and have the default dialog do the SaveAs?

Reply:
> If I set Cancel=true but don't set SuppressDefaultDialog, will the default dialog show up to try its own operation or will the Cancel win?

Cancel should win in this case

> If I "handle" the SaveAs request and populate args Kind and ResultFilePath, but don't set SuppressDefaultDialog=true, does the system ignore my other args values and have the default dialog do the SaveAs?

Reasonable options: 
 1 suppressdefaultdialog is false so args are ignored and default dialog is shown.
 2 Or suppressdefaultdialog is false so some args are applied to the default dialog's defaults.

We choose (1) and please ensure this is documented.